tests/components/devolo_home_network/test_switch.py

```diff
@@ -127,36 +124,31 @@ async def test_update_enable_guest_wifi(
     mock_device.device.async_get_wifi_guest_access.return_value = WifiGuestAccessGet(
         enabled=True
     )
-    with patch(
-        "devolo_plc_api.device_api.deviceapi.DeviceApi.async_set_wifi_guest_access",
-        new=AsyncMock(),
-    ) as turn_on:
-        await hass.services.async_call(
-            PLATFORM, SERVICE_TURN_ON, {"entity_id": state_key}, blocking=True
-        )
+    await hass.services.async_call(
+        PLATFORM, SERVICE_TURN_ON, {"entity_id": state_key}, blocking=True
+    )
 
-        state = hass.states.get(state_key)
-        assert state is not None
-        assert state.state == STATE_ON
-        turn_on.assert_called_once_with(True)
+    state = hass.states.get(state_key)
+    assert state is not None
+    assert state.state == STATE_ON
+    mock_device.device.async_set_wifi_guest_access.assert_called_once_with(True)
+    mock_device.device.async_set_wifi_guest_access.reset_mock()
 
     freezer.tick(REQUEST_REFRESH_DEFAULT_COOLDOWN)
     async_fire_time_changed(hass)
     await hass.async_block_till_done()
 
     # Device unavailable
     mock_device.device.async_get_wifi_guest_access.side_effect = DeviceUnavailable()
-    with patch(
-        "devolo_plc_api.device_api.deviceapi.DeviceApi.async_set_wifi_guest_access",
-        side_effect=DeviceUnavailable,
-    ):
+    mock_device.device.async_set_wifi_guest_access.side_effect = DeviceUnavailable()
+
+    with pytest.raises(HomeAssistantError):
```

Comment:
```suggestion
    with pytest.raises(HomeAssistantError, match=".."):
```
We can match on the exception string